### PR TITLE
test: don't redirect stdout/stderr to console

### PR DESCRIPTION
**Pull request by mrc0mmand**
Turns out that redirecting a lot of output to the console can have some
funny effects, like random kernel soft lockups. I spotted this in
various CIs, but it remained almost entirely hidden thanks to
`softlockup_panic=1`, until https://github.com/systemd/systemd/commit/1a36d2672f32fbbc695e77e12dfa43b646243d63 which introduced a couple of
tests that log quite a lot in a short amount of time. This, in
combination with newer kernel version, which, for some reason, seem to
be more susceptible to such soft lockups, made the Arch Linux jobs soft
lockup quite a lot, see [0].

While debugging this I also noticed that runs which don't redirect
stdout/stderr to the console are noticeably faster, e.g.:
```
    # TEST-71 nspawn + QEMU (KVM), StandardOutput=journal+console
    Elapsed (wall clock) time (h:mm:ss or m:ss): 0:24.64
    # TEST-71 nspawn + QEMU (KVM), StandardOutput=journal
    Elapsed (wall clock) time (h:mm:ss or m:ss): 0:17.95

    # TEST-71 nspawn + QEMU, StandardOutput=journal+console
    Elapsed (wall clock) time (h:mm:ss or m:ss): 2:04.70
    # TEST-71 nspawn + QEMU, StandardOutput=journal
    Elapsed (wall clock) time (h:mm:ss or m:ss): 1:44.48

    # TEST-04 QEMU, StandardOutput=journal+console
    Elapsed (wall clock) time (h:mm:ss or m:ss): 4:22.70
    # TEST-04 QEMU, StandardOutput=console
    Elapsed (wall clock) time (h:mm:ss or m:ss): 5:04.67
```
Given all this, let's effectively revert https://github.com/systemd/systemd/commit/ba7abf79a5a2df2a93332fc32e8e3c268c74abc7, and show the last
50 (or 100, if the test fails) lines from the journal instead when saving it.
This should, hopefully, address both issues while still preserving at least
some level of debug-ability of the previous solution.

Resolves: https://github.com/systemd/systemd-centos-ci/issues/660

[0] https://github.com/systemd/systemd-centos-ci/issues/660

### CODE REVIEW

**Comment by mrc0mmand** [comment]
I think I can say that this indeed helps with systemd/systemd-centos-ci#660 - I had the reproducer running in a loop for 3+ hours and it hasn't crashed once, whereas before it took just a couple of tries. I still want to do a couple more tests though, since there was a strange lockup in TEST-64 as well.

In the meantime, @bluca, would this be acceptable from the Ubuntu CI side? As this effectively drops the `$TEST_SHOW_JOURNAL` variable and the corresponding output (which was a bit useless anyway if you ask me), and instead it dumps (hopefully) relevant lines after the test finishes, i.e.:

```
[  OK  ] Finished systemd-poweroff.service.
[  OK  ] Reached target poweroff.target.
Sending SIGTERM to remaining processes...
Sending SIGKILL to remaining processes...
All filesystems, swaps, loop devices, MD devices and DM devices detached.
Powering off.
Container TEST-74 has been shut down.
 --- Last 50 messages from /var/tmp/systemd-test-TEST-74-AUX-UTILS/root/var/log/journal ---
[ 1255.962532] testsuite-74.sh[1259]: + varlinkctl list-interfaces
[ 1255.967033] varlinkctl[1260]: Too few arguments.
[ 1255.968556] testsuite-74.sh[1261]: + varlinkctl list-interfaces ''
[ 1255.972530] varlinkctl[1262]: Failed to connect to URL '': Protocol not supported
[ 1255.974355] testsuite-74.sh[1263]: + varlinkctl introspect
[ 1255.978555] varlinkctl[1264]: Too few arguments.
[ 1255.980175] testsuite-74.sh[1265]: + varlinkctl introspect /run/systemd/journal/io.systemd.journal
[ 1255.984211] varlinkctl[1266]: Too few arguments.
...
[ 1256.040785] testsuite-74.sh[1285]: + set +x
[ 1256.040971] testsuite-74.sh[1285]: PASSED TESTS:  17:
[ 1256.040971] testsuite-74.sh[1285]: ------------------
[ 1256.040971] testsuite-74.sh[1285]: /usr/lib/systemd/tests/testdata/units/testsuite-74.battery-check.sh
[ 1256.040971] testsuite-74.sh[1285]: /usr/lib/systemd/tests/testdata/units/testsuite-74.busctl.sh
[ 1256.040971] testsuite-74.sh[1285]: /usr/lib/systemd/tests/testdata/units/testsuite-74.cgls.sh
[ 1256.040971] testsuite-74.sh[1285]: /usr/lib/systemd/tests/testdata/units/testsuite-74.cgtop.sh
[ 1256.040971] testsuite-74.sh[1285]: /usr/lib/systemd/tests/testdata/units/testsuite-74.coredump.sh
[ 1256.041116] testsuite-74.sh[1285]: /usr/lib/systemd/tests/testdata/units/testsuite-74.delta.sh
[ 1256.041116] testsuite-74.sh[1285]: /usr/lib/systemd/tests/testdata/units/testsuite-74.escape.sh
[ 1256.041116] testsuite-74.sh[1285]: /usr/lib/systemd/tests/testdata/units/testsuite-74.firstboot.sh
[ 1256.041116] testsuite-74.sh[1285]: /usr/lib/systemd/tests/testdata/units/testsuite-74.id128.sh
[ 1256.041116] testsuite-74.sh[1285]: /usr/lib/systemd/tests/testdata/units/testsuite-74.machine-id-setup.sh
[ 1256.041116] testsuite-74.sh[1285]: /usr/lib/systemd/tests/testdata/units/testsuite-74.modules-load.sh
[ 1256.041116] testsuite-74.sh[1285]: /usr/lib/systemd/tests/testdata/units/testsuite-74.mount.sh
[ 1256.041116] testsuite-74.sh[1285]: /usr/lib/systemd/tests/testdata/units/testsuite-74.networkctl.sh
[ 1256.041116] testsuite-74.sh[1285]: /usr/lib/systemd/tests/testdata/units/testsuite-74.path.sh
[ 1256.041116] testsuite-74.sh[1285]: /usr/lib/systemd/tests/testdata/units/testsuite-74.pstore.sh
[ 1256.041116] testsuite-74.sh[1285]: /usr/lib/systemd/tests/testdata/units/testsuite-74.run.sh
[ 1256.041116] testsuite-74.sh[1285]: /usr/lib/systemd/tests/testdata/units/testsuite-74.varlinkctl.sh
[ 1256.041399] testsuite-74.sh[43]: + touch /testok
Spawning getter /systemd-meson-build/journalctl -o export -D /var/tmp/systemd-test-TEST-74-AUX-UTILS/root/var/log/journal...
Finishing after writing 23359 entries
-rw-r----- 1 root root 50331648 Nov 21 14:51 /var/tmp/systemd-test-TEST-74-AUX-UTILS/system.journal
TEST-74-AUX-UTILS RUN: Tests for auxiliary utilities [OK]
```

**Comment by bluca** [comment]
mmh I do use the output in the log to quickly spot whether an issue is new or old - I could easily enable it explicitly over there with a variable though if we want it off by default

**Comment by mrc0mmand** [comment]
> mmh I do use the output in the log to quickly spot whether an issue is new or old - I could easily enable it explicitly over there with a variable though if we want it off by default

The logs would be still there - it always dumps the last 50 lines of journal, or 100 in case the test fails:
https://github.com/systemd/systemd/pull/30119/files#diff-62348290635aedb88a6451b5fdc798074172aeb94a360599215141224f329081R1734-R1739

It just does that _after_ the test, so the logs don't go through the slow console.

On a similar note - is there any way how to access the logs of finished Ubuntu CI jobs? According to the queue only the amd64 jobs is not finished yet, but I can't find the links to the artifacts anywhere.

**Comment by bluca** [comment]
They are stored somewhere accessible, but you need to know a magic random id from the queued job, and I have no idea how to get that, so until they fix the reporting back it's anyone's guess

**Comment by mrc0mmand** [comment]
This seems to help with the TEST-64 soft lockups as well. Let's wait for Ubuntu CI to come back alive, though, just to see how it'll look there before moving further.

**Comment by mrc0mmand** [comment]
> I tweaked the logic a bit, so now we dump all testsuite-related messages if the test fails, otherwise dump only the last 50 messages. This should make debugging fails still as "easy" as before and also make the final log a bit smaller.

Nah, scratch that. Let's always dump all testsuite-related logs, since I completely forgot I use them quite a lot when checking if the newly added tests are actually run at all. So now there shouldn't be any "regression" when it comes to debugability, we just avoid using the slow console to show the logs by dumping the logs after the test finishes.

This should, hopefully, finally unfuck the Arch jobs wrt https://github.com/systemd/systemd-centos-ci/issues/660, so I can switch the jobs to the latest kernel (instead of the 6.1-lts) before the final release.

File: test/test-functions [hunk 1]
```diff
@@ -1032,7 +1032,6 @@ EOF
 DefaultEnvironment=${default_environment[*]}
```
Comment (yuwata) [inline test/test-functions:1032]: commit message: cannot follow the commit hash ba7abf79a5x

**Review by yuwata — APPROVED** [review]

File: test/test-functions (hunk 1 above)
Comment (mrc0mmand) [reply test/test-functions:1032]: Whoops, thank you.

**Pull request merged**